File: examples/checks/photosynthetic_exciton_transfer.py
# Independent Python checks for the photosynthetic_exciton_transfer example.
from .common import run_fragment_checks

CHECKS = [
    ('the tuned complex can sample exciton pathways coherently', ['The tuned complex combines strong excitonic coupling, delocalization, a tuned vibronic bridge, moderate dephasing, short-lived coherence, and a downhill route to the reaction cente']),
    ('the tuned complex can use vibronically assisted transfer', ['YES for the tuned antenna complex.']),
    ('short-lived quantum assistance is enough in the tuned downhill regime', ['The detuned contrast complex has weak coupling, absent delocalization, no vibronic bridge, strong dephasing, and a trapping mismatch, so the same efficient delivery task is blocked']),
    ('efficient exciton transfer is possible in the tuned complex', ['NO for the detuned, strongly decohered contrast complex.']),
    ('the tuned complex can deliver excitation to the reaction center', ['## Answer']),
    ('the detuned complex cannot sample pathways coherently', ['## Answer']),
    ('the detuned complex cannot use vibronically assisted transfer', ['## Answer']),
    ('the detuned complex cannot achieve directed reaction-center transfer', ['## Answer']),
    ('the detuned complex cannot achieve efficient exciton transfer', ['## Answer']),
    ('the detuned complex cannot deliver excitation efficiently to the reaction center', ['## Answer']),
]

def run(ctx):
    return run_fragment_checks(ctx, CHECKS)

File: examples/checks/photosynthetic_exciton_transfer.go
package main

func checkPhoto(ctx *Context) []Check {
	d := ctx.M()
	comps := map[string]anymap{}
	for _, c := range maps(d["complexes"]) {
		comps[str(c["role"])] = c
	}
	tuned := comps["positive"]
	detuned := comps["contrast"]
	rc := str(d["reactionCenter"])
	tunedOK := photoEff(tuned, rc)
	detunedOK := photoEff(detuned, rc)
	return []Check{{"the tuned complex has strong excitonic coupling", str(tuned["excitonCoupling"]) == "Strong"}, {"the tuned complex has delocalized states", str(tuned["stateDelocalization"]) == "Present"}, {"the tuned complex has a tuned vibronic bridge", str(tuned["vibronicBridge"]) == "Tuned"}, {"the tuned complex has moderate dephasing and short-lived coherence", str(tuned["dephasing"]) == "Moderate" && str(tuned["electronicCoherenceLifetime"]) == "Short"}, {"the tuned complex is connected downhill to the reaction center", str(tuned["energyLandscape"]) == "DownhillToReactionCenter" && str(tuned["connectedTo"]) == rc}, {"efficient transfer is independently derived for the tuned complex", tunedOK && contains(ctx.Answer, "YES for the tuned antenna complex")}, {"the detuned complex lacks strong coupling, delocalization, and a vibronic bridge", str(detuned["excitonCoupling"]) == "Weak" && str(detuned["stateDelocalization"]) == "Absent" && str(detuned["vibronicBridge"]) == "Absent"}, {"the detuned complex has strong dephasing and a trapping mismatch", str(detuned["dephasing"]) == "Strong" && str(detuned["energyLandscape"]) == "TrappingMismatch"}, {"efficient delivery is blocked for the detuned contrast complex", !detunedOK && contains(ctx.Answer, "NO for the detuned")}, {"the reaction-center connection alone is insufficient without the other conditions", str(detuned["connectedTo"]) == rc && !detunedOK}}
}
